packages/artifact/src/internal/download/download-artifact.ts

```diff
@@ -140,13 +144,16 @@ export async function downloadArtifactPublic(
 
   try {
     core.info(`Starting download of artifact to: ${downloadPath}`)
-    await streamExtract(location, downloadPath)
-    core.info(`Artifact download completed successfully.`)
+    if (await streamExtract(location, downloadPath)) {
+      core.info(`Artifact download completed successfully.`)
+      return {downloadPath, skipped: false}
+    } else {
+      core.info(`Artifact download skipped.`)
```

Comment:
If a user requests a specific artifact to download and that artifact can't be downloaded, why should that silently fail?

Shouldn't that be an error?

Reply:
I'm reluctant to change the default behavior here as it could be a breaking change for users.

Reply:
iiuc the API only expects zip content type for extraction when used with `actions/download-artifact@v4`: https://github.com/actions/toolkit/blob/bb2278e5cfbb40afc20890c415e9ffa836631cd5/packages/artifact/src/internal/download/download-artifact.ts#L92

And it [currently breaks](https://github.com/actions/download-artifact/issues/328#issuecomment-2194122935) workflows when artifacts with other content-type are being downloaded.

I guess we could create an error type is content type does not match so `actions/download-artifact` can catch it?

Reply:
I thought silently skipping unrelated artifacts not uploaded with `actions/upload-artifact` would be best so it doesn't require any changes in `actions/download-artifact`.